packages/flutter_tools/lib/src/macos/macos_ipad_device.dart

```diff
@@ -0,0 +1,147 @@
+          processManager: processManager,
+          logger: logger,
+          fileSystem: fileSystem,
+          operatingSystemUtils: operatingSystemUtils,
+        );
+
+  final OperatingSystemUtils _operatingSystemUtils;
+
+  @override
+  String get name => 'Mac Designed for iPad';
+
+  @override
+  Future<TargetPlatform> get targetPlatform async => TargetPlatform.darwin;
+
+  @override
+  bool isSupported() => _operatingSystemUtils.hostPlatform == HostPlatform.darwin_arm;
+
+  @override
+  bool isSupportedForProject(FlutterProject flutterProject) {
+    return flutterProject.ios.existsSync() && _operatingSystemUtils.hostPlatform == HostPlatform.darwin_arm;
+  }
+
+  @override
+  String executablePathForDevice(ApplicationPackage package, BuildMode buildMode) => null;
+
+  @override
+  Future<LaunchResult> startApp(
+    IOSApp package, {
+    String mainPath,
+    String route,
+    @required DebuggingOptions debuggingOptions,
+    Map<String, dynamic> platformArgs = const <String, dynamic>{},
+    bool prebuiltApplication = false,
+    bool ipv6 = false,
+    String userIdentifier,
+  }) async {
+    // Only attaching to a running app launched from Xcode is supported.
+    throw UnimplementedError('Building for "$name" is not supported.');
+  }
+
+  @override
+  Future<bool> stopApp(
+    IOSApp app, {
+    String userIdentifier,
+  }) async => false;
+
+  @override
+  Future<void> buildForDevice(
+    covariant IOSApp package, {
+    String mainPath,
+    BuildInfo buildInfo,
+  }) async {
+    // Only attaching to a running app launched from Xcode is supported.
+    throw UnimplementedError('Building for "$name" is not supported.');
+  }
+}
+
+class MacOSDesignedForIPadDevices extends PollingDeviceDiscovery {
+  MacOSDesignedForIPadDevices({
+    @required Platform platform,
+    @required IOSWorkflow iosWorkflow,
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required FileSystem fileSystem,
+    @required OperatingSystemUtils operatingSystemUtils,
+  })  : _logger = logger,
+        _platform = platform,
+        _iosWorkflow = iosWorkflow,
+        _processManager = processManager,
+        _fileSystem = fileSystem,
+        _operatingSystemUtils = operatingSystemUtils,
+        super('Mac designed for iPad devices');
+
+  final IOSWorkflow _iosWorkflow;
+  final Platform _platform;
+  final ProcessManager _processManager;
+  final Logger _logger;
+  final FileSystem _fileSystem;
+  final OperatingSystemUtils _operatingSystemUtils;
+
+  @override
+  bool get supportsPlatform => _platform.isMacOS;
+
+  /// iOS (not desktop macOS) development is enabled, the host is an ARM Mac,
+  /// and discovery is allowed for this command.
+  @override
+  bool get canListAnything =>
+      _iosWorkflow.canListDevices && _operatingSystemUtils.hostPlatform == HostPlatform.darwin_arm && allowDiscovery;
+
+  /// Set to show ARM macOS as an iOS device target.
+  static bool allowDiscovery = false;
```

Comment:
I couldn't figure out a non-global way to set this (modeled on `FlutterTesterDevices.showFlutterTesterDevice`).  I tried to make it a property on a context override on `IOSWorkflow` but I would have needed to set it the context override only for the `attach` command in: 
https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L232-L235
and
https://github.com/flutter/flutter/blob/80668d20f86be0fefd9f210db0d23c4bb2d68d59/packages/flutter_tools/lib/src/commands/attach.dart#L178-L179
and
https://github.com/flutter/flutter/blob/80668d20f86be0fefd9f210db0d23c4bb2d68d59/packages/flutter_tools/lib/src/commands/attach.dart#L210-L213

Welcome suggestions to make this better.

Reply:
static is fine, please no more context overrides